wsclient/src/main/java/org/hpccsystems/ws/client/utils/Utils.java

```diff
@@ -1261,4 +1262,47 @@ static public String getTraceParentHeader(Span span)
 
         return traceparent;
     }
+
+    static public boolean containsVMArgument(String vmArgName)
+    {
+        List<String> argslist = ManagementFactory.getRuntimeMXBean().getInputArguments();
+        for (String string : argslist)
+        {
+            if(string.matches("(?i)(" + vmArgName + "):.*"))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    static public String fetchVMArgument(String vmArgName)
+    {
+        List<String> argslist = ManagementFactory.getRuntimeMXBean().getInputArguments();
+        for (String string : argslist)
+        {
+            if(string.matches("(?i)(" + vmArgName + "):.*"))
+            {
+                String[] keyval = string.split(vmArgName+":");
+
+                return keyval[1];
+            }
+        }
+        return "";
+    }
+
+    static public boolean isOtelJavaagentUsed()
```

Comment:
> Should this be called isJavaagentUsed() since we aren't explicitly checking if it is the Otel java agent?

Take a look at L:1300, the intent there is to ensure the -javaagent argument includes the opentelemetry name as you originally suggested. Perhaps we should create a more detailed regex to ensure the filename is opentelemetry-javaagent.jar?

Reply:
Oh yeah totally missed that.